### ISSUE TITLE
ToC inserted as unprocessed markdown code now

### ISSUE THREAD

**Issue by RandyHarr** [jordan2175/markdown-tools#15]
Inserting the ToC goes in as unprocessed markdown code. Starting with "#Table of Contents" and proceeding with "- [1 header 1..."

**Comment by kinshadow** [comment]
This bug seems to be 2 years old.  Is there a workaround for this?

**Comment by jordan2175** [comment]
I am not sure what the problem is here. The markdown TOC works.

**Comment by RandyHarr** [comment]
Still broken for me on Chrome 136.0.7103.93 using Markdown Tools 4.6. I have been able to start with an old document with a working table of contents and edit it to a new document to bring over a ToC.  That seems to still work for now.

**Comment by jordan2175** [comment]
I just tested this in Chrome Version 136.0.7103.93 (Official Build) (arm64) and it works fine.  See screenshot.

![Image](https://github.com/user-attachments/assets/ae3ee1ca-1509-4f3c-b120-cfa4081f09bd)

**Comment by kinshadow** [comment]
Yes, but the markdown does not get processed by Google Docs.  It behaves as though it was pasted in.  Are you doing something else to get the document to process the formatting?

**Comment by jordan2175** [comment]
Please help me understand what you mean. I create a Google Doc, I use the Google Docs feature to make the section headers either Level 1 or Level 2 or Level 3 etc. Once that is done I can turn them into markdown headings. Basically with the right number of # signs. From there I can generate a TOC.

**Comment by jordan2175** [comment]
Then if you want to save this as an .md file and push the content to Github you can use the save as plain-text feature in Google docs. Or you can just copy and paste into your favorite plain-text IDE / editor. I use this all the time and this is why I created this pluging.

**Comment by kinshadow** [comment]
I'm not pushing to GitHub.  I'm just using this feature in Google Docs to make documents.   I'd like Google Docs to use the markdown to format the TOC correctly or an option to just have it formatted correctly without the markdown.  Typing markdown into Google Docs will normally just interpret it on the fly format the text.  Pasting (or using this tool) only gives markdown text and I don't see anyway to make Google Docs process the markdown from this extension.

**Comment by jordan2175** [comment]
Oh, I understand now. When I wrote this tool, Google docs had no support for Markdown at all. Also, my primary use case was to automatically add section numbers to headings along with the correct number of # symbols for markdown. After I got all of that working, I decided to add support for generating a TOC. All of this stemmed from documents that I write in Google Docs that I then need to make markdown versions of. I like doing it this way, so that I can use pretty formatting for when I am working in the document, like different sizes for headings and colors for the fonts and bold or not bold. As all of these are tied to the "heading" level in Google Docs. But then I can easily render this as Markdown. I have done a lot of straight markdown in documents that are around 5-10 pages. But most of my documents that I use this for are 400-800 pages in length. So being able to have some better visual aspects, and the ability to use Google Docs outline features was important to me.

So I know this does not help your use case, as you are trying to have it create a TOC based on existing markdown. This tool generates a markdown TOC based on actual headings. Now it might be possible to first convert your markdown headings into actual Google Doc headings first, so that under the covers they are real Google Doc headings. But I would have to look into that.

**Comment by RandyHarr** [comment]
I exclusively use the extension within Google Docs.  Never use or write markdown code (directly).  Before, the extension would put in the hyperlinked / processed ToC directly into the document.  Now it puts the markdown code that never gets interpreted.  I had not been aware that Google Docs may be using markdown behind the scenes till your mention above. Sounds like @kinshadow may be having the same experience I am.  Not sure what changed in Google Docs two years ago. Maybe a default setting in Google Docs somewhere?

I have tried to use the extension "Markdown Converter" to process the inserted ToC.  But it does not work except on text formatting. The default Google Docs ToC does not index subsections.

See https://bit.ly/3nJqxqo for a doc where the ToC is processed (page 4) the way it used to work.  See https://bit.ly/3U6eYX8 for one for how it behaves today as you show above.  Note that I use the extension to insert just "Heading Numbers" and not "Heading Numbers (Markdown)". That in combination with using the Google Docs ToC with manual post processing gets one closer to the original result.
